### ISSUE TITLE
Incremental build of SDK projects fails when files are removed from globs

### ISSUE THREAD

**Issue by rladuca** [microsoft/msbuild#4963]
### Steps to reproduce

1. Create a  C# WPF Core application.
2. Add a WPF User Control to the project.
3. Build the project successfully.
4. Delete the User Control you just added.
5. Build the project.

You can also do the same with .NET Core Console applications

1. Create a  C# Console Core application.
2. Add a new class to the project (in a new CS file)
3. Build the project successfully and verify the class exists in the binary.
4. Delete the cs file you just added
5. Build the project.

### Expected  behavior

In WPF, the project should build successfully and the deleted UserControl should not be in the binary.
In Console, the project should build successfully and the deleted class should not be in the binary.

### Actual behavior
In WPF, MarkupCompile fails due to trying to compile a file that is no longer on disk.
In Console, the resulting binary still contains the removed type.

### Environment data
```
Microsoft (R) Build Engine version 16.5.0-preview-19570-01+9c33693c4 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

16.5.0.57001
```

Related to #701 and likely a regression from #4482 (@rainersigwald )

**Comment by rladuca** [comment]
FYI @sambent @vatsan-madhavan

**Comment by BenVillalobos** [comment]
I was able to repro both cases, but the regression in #4482 only applied to the C# case. The fix for the C# case is in #4980.

@rladuca I believe the WPF case is a separate issue that's not related to #4482. 

An interesting note is that the build will succeed when building after it fails. _Something_ is telling the target `MarkupCompilePass1` that `abc.xaml` is an input despite being deleted just before building.

**Comment by rladuca** [comment]
@BenVillalobos WPF took a fix in https://github.com/dotnet/wpf/issues/2287, thanks for the turnaround!
